src/xbot/common/logging/PeriodicDeltaObserver.java

```diff
@@ -0,0 +1,36 @@
+package xbot.common.logging;
+
+import org.apache.log4j.Logger;
+
+public class PeriodicDeltaObserver {
+
+    protected static Logger log = Logger.getLogger(PeriodicDeltaObserver.class);
+    private final String callerName;
+    private String message;
+    
+    private double checkPowerThreshold;
+    
+    private double oldPower = 0;
+    
+    public PeriodicDeltaObserver(String callerName, String message, double checkPowerThreshold) {
+        this.callerName = callerName;
+        this.message = message;
+        this.checkPowerThreshold = checkPowerThreshold;
+    }
+    
+    public void setMessage(String message) {
+        this.message = message;
+    }
+    
+    public void setCheckPowerThreshold(double power) {
+        this.checkPowerThreshold = power;
+    }
+    
+    public void checkPower(double newPower) {
+        if (Math.abs(oldPower - newPower) > checkPowerThreshold) {
+            log.info("From " + callerName + ": " + message);
+        }
+        oldPower = newPower;
```

Comment:
If we didn't notify, we don't want to replace the power value. We want to notify every time the value has changed by more than the threshold _since the last time the caller was notified_.